docs/testing/stores.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: docs
+title: Testing Stores
+description: How to test alt stores
+permalink: /docs/testing/stores/
+---
+
+# Testing Stores
+
+## Conceptual how to
+
+The tricky thing about testing alt stores is that alt wraps up the store class we create into an `AltStore`, which essentially adds alt specific functionality (`getState`, `emitChange`, etc), while hiding the implementation and data in our store. The only way we can interact with our store once alt wraps it is via the methods alt adds like `getState` as well as any static methods we defined in our class. Alt does this to enforce the unidirectional flow of data and prevent views from updating stores. Flux specifies that stores be updated via actions only. This means testing our static methods is easy, but testing our other methods is not as straightforward.
+
+The primary functionality of Stores in the Flux pattern is to listen for Actions, update/manage data to be used by the view, and emit change events to let the views know that data has been changed and they need to update. Based on this, the main thing we want to is dispatch an event that our store is listening to (via `alt.dispatcher.dispatch(payload)`) and then check if the data in the store is updated in the way we expect (via methods that return data like `store.getState()` or `store.myStaticMethod()`).
+
+This form of "blackbox testing" should cover most of your store testing needs, but what if our store methods that respond to actions do other things besides update data, or what if we have some other non-static helper methods that we need to test?
+
+One thing we can do is to test our class before it is wrapped by alt, the unwrapped class. We should trust the alt library to test its own internals, so we can just test the inaccessible methods in our plain class by exporting it separately from our alt wrapped store (via `alt.createStore`). To do this, we can export it as a separate file, or use a named export for our unwrapped class and the default export for our alt wrapped class (in the example below, I will use the latter for simplicity).
+
+Another method to get at the unwrapped class is to do something like `AltWrappedStore[Object.getOwnPropertySymbols(AltWrappedStore)[2]]` as detailed [here](https://gist.github.com/troutowicz/d9dcb7e7e9e77032f602).
```

Comment:
We should remove this, while it's clever it's not recommended since the symbol may change indices in the future.